modules/order/src/Entity/OrderItem.php

```diff
@@ -108,6 +110,26 @@ public function getQuantity() {
   /**
    * {@inheritdoc}
    */
+  public function getItemsQuantity() {
+    $settings = $this->getQuantityWidgetSettings();
+    $step = isset($settings['#step']) && is_numeric($settings['#step']) ? $settings['#step'] + 0 : 1;
+    $quantity = $this->getQuantity();
+    return (string) is_int($step) ? $quantity : (is_float($step) && $quantity > 0 ? '1' : $quantity);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuantityWidgetSettings() {
```

Comment:
I agree with you but in our example getQuantityWidgetSettings() is only exposed centralized functionality. Without this method exposed every class operating on order item objects need actually to duplicate the method if they need to check widget settings.

Well, to comply with API requirements could we define a separate service?